pkg/configurableinterpreter/rule.go

```diff
@@ -1,4 +1,4 @@
-package interpret
+package configurableinterpreter
```

Comment:
How about moving it down into the `pkg/util` package?

Besides, some type of struct (such as `ruleArgs` and `ruleResult`) may be needed to export, it will be used in #2824

Reply:
> How about moving it down into the `pkg/util` package?

I prefer `pkg/util/configurableinterpreter`, otherwise the name `Rule` is too simple, developers are confused: What rule?  Unless rename to `ConfigurableInterpreterRule`,  `ConfigurableInterpreterRuleArgs` ...

Reply:
> Besides, some type of struct (such as ruleArgs and ruleResult) may be needed to export, it will be used in https://github.com/karmada-io/karmada/pull/2824



I think this problem is not the concern of this PR, and we can solve it in other PRs.

Reply:
> This is a global switch. You need to set the a field of the policy to true to enable the follow-up distribution of corresponding resources.

This is a concern because this change may conflict with the timing of #2824's merger.

How about changing the file directory first?

Reply:
done!